### DISCUSSION TITLE
[Feature] Ability to toggle audio effect and tune it's severity

### DISCUSSION THREAD

**Discussion by HossBigft** [BudgetAirpods/InsanityRemastered#18] [Ideas]
I would like to be able to toggle or even tune how strong echo effect. Right now, if you are solo, you just lose ability to hear after some time inside

**Comment by BudgetAirpods** [comment]
You can turn it and the visual effect that accompanies it off in the config. The setting should be under the hallucinations section.
(I'm aware the settings are a mess right now and I will fix this.)

I'll see if I can make it to where you can change how strong it will be.

**Comment by HossBigft** [comment]
Thanks for quick reply, I tried to disable Able to experience auditory hallucinations, but it seems to affect only fake sounds of enemies.

Also I would like way to change how often any type of hallucination can happen. For example, since I installed the mod I saw visual hallucination only once and it was fake player.

After last update I tried to see what hallucinations looks like and set Time between Hallucination RNG to 1. I started to hear enemies pretty often but that's was it.

**Comment by HossBigft** [comment]
Here is  our [config](https://github.com/demosche/lethal-company-modded/blob/main/ModdedBuild/BepInEx/config/BudgetAirpods.InsanityRemastered.cfg)

**Comment by BudgetAirpods** [comment]
> Also I would like way to change how often any type of hallucination can happen. For example, since I installed the mod I saw visual hallucination only once and it was fake player.
> 
> After last update I tried to see what hallucinations looks like and set Time between Hallucination RNG to 1. I started to hear enemies pretty often but that's was it.

I've thought about having some kind of rarity system for each hallucination. However, I have made most if not all hallucinations occur at any point inside the facility (this is in the unreleased update I'm working on).

As for your second point, the reason behind this is because I coded it for certain hallucinations to only occur when you have a high enough sanity level. (For example, the fake player would only spawn at 50% full insanity cap. The auditory hallucinations and fake items were/are the only ones that should occur at the lowest level. Again, this has been fixed in the update I'm working on.

**Comment by BudgetAirpods** [comment]
> I've thought about having some kind of rarity system for each hallucination. However, I have made most if not all hallucinations occur at any point inside the facility (this is in the unreleased update I'm working on).

To give more examples of this new system: The fake player can spawn at any sanity level now. This _should_ make it more common

The power loss event however has been changed to occur only at the highest sanity level.

If you want to try it out, the alpha version of this update is [available for testing, along with the changelog](https://github.com/BudgetAirpods/InsanityRemastered/tree/(ALPHA/EXPERIMENTAL)-Update-V1.1). I really wouldn't recommend it for normal playthroughs though.

**Comment by HossBigft** [comment]
> > Also I would like way to change how often any type of hallucination can happen. For example, since I installed the mod I saw visual hallucination only once and it was fake player.
> > After last update I tried to see what hallucinations looks like and set Time between Hallucination RNG to 1. I started to hear enemies pretty often but that's was it.
> 
> I've thought about having some kind of rarity system for each hallucination. However, I have made most if not all hallucinations occur at any point inside the facility (this is in the unreleased update I'm working on).
> 
> As for your second point, the reason behind this is because I coded it for certain hallucinations to only occur when you have a high enough sanity level. (For example, the fake player would only spawn at 50% full insanity cap. The auditory hallucinations and fake items were/are the only ones that should occur at the lowest level. Again, this has been fixed in the update I'm working on.

Great to see, thanks for your work. When I was testing mod, I set solo  insanity growth to 5, so I lost it pretty fast and there was nothing but sound. Even before last update when I was barely able to see and hear, there was no visual hallucinations of players of enemies.

**Reply by BudgetAirpods** [reply]
Hallucinations of enemies have been scrapped, sadly. 

There are too many variables I would have to account for. (Animations, sound effects, storing the model and making it work properly as its own AI, etc.).

**Comment by HossBigft** [comment]
I played 1.0.8 with my brother and in my experience walkie talkies talks too much to the point we had troubles communicating. They were talking even in space if we didn't shut them off.
Also it feels like someone is talking next to you rather over radio because skinwalker voicelines lacks radio filters and beeps. We tried to disable it but the only related option I found is audio hallucinations and disabling them did nothing. Our [config](https://github.com/demosche/lethal-company-modded/blob/main/ModdedBuild/BepInEx/config/BudgetAirpods.InsanityRemastered.cfg).

I also disabled both notifications and messages but still got increased heartrate popup.

Talking radio generated a few cool moments, though, when it was talking in hand of other player, so from side it was looking like player is talking, when he infact doesn't. So maybe you could make it so radio would talk only in hands of other players but not when you holding it? Or maybe make players play skinwalker lines, so it would create confusion.

It would be great if there was a way to toggle what hallucinations can happen and at which chance because since installing this mod I saw fake player exactly once and can't make him appear no matter how I crank up the config file.

**Comment by HossBigft** [comment]
@BudgetAirpods so how do I shut them up? It's hard to use radio when it talks over me on both ends
